snap/Dockerfile.build

```diff
@@ -1,16 +1,36 @@
 FROM ubuntu:16.04
 
-# this is basically the same as the amd64 upstream:
-# https://github.com/snapcore/snapcraft/blob/master/Dockerfile
+# allow specifying the architecture from the build arg command line
+ARG ARCH
+
+# this is essentially the same as the upstream dockerfile
+# here: https://github.com/snapcore/snapcraft/blob/master/docker/stable.Dockerfile
+# except we also specify the architecture to download so that this works
+# on other architectures
 RUN apt-get update && \
   apt-get dist-upgrade --yes && \
   apt-get install --yes \
-  git \
-  snapcraft \
-  && \
+  curl sudo jq squashfs-tools && \
```

Comment:
I realize this logic is cut & pasted from the upstream dockerfile, but it might be helpful to add a short bullet list that describes each step below.

It looks to me like you're querying the snap details for both core and snapcraft, downloading both, and unsquashing both under /snap/<name>/current, which is a hacky way to install snapcraft w/out requiring that snapd run in the snap...

Reply:
Yes it is a very hacky way to do this, but it's the official upstream way the docker image works, so I think it's okay as long as we document it and explain what and why we're doing it. I will add more comments explaining what we're doing and why though